### PR TITLE
perf: replace divan with criterion

### PR DESCRIPTION
**Pull request by anonrig**
Replaces Divan with Criterion. This opens the possibility of integration codspeed.io into our workflow.

### CODE REVIEW

File: rust_snuba/benches/processors.rs [hunk 1]
```diff
@@ -176,199 +102,94 @@ fn create_stream_processor(
...
+                    black_box(MSG_COUNT),
+                ));
+                loop {
+                    let res = processor.run_once();
+                    if matches!(res, Err(RunError::Poll(ConsumerError::EndOfPartition))) {
+                        // FIXME: this pretty much means that we *polled* the partition to the end,
+                        // it does not mean that we actually *committed* everything
+                        // (aka we finished actually processing everything).
+                        break;
+                    }
```
Comment (Swatinem) [inline rust_snuba/benches/processors.rs:131]: I believe you can also properly `close` the processor now, as the underlying deadlock was fixed.

**Review by Swatinem — APPROVED** [review]
lgtm. it would be nice to add back the throughput measurement/output as well.

**Comment by anonrig** [comment]
> lgtm. it would be nice to add back the throughput measurement/output as well.

Should I go for `KiB/s` or `Kelem/s`?

File: rust_snuba/benches/processors.rs (hunk 1 above)
Comment (anonrig) [reply rust_snuba/benches/processors.rs:131]: Should I just call `processor.shutdown()` on EndOfPartition error, and break?

**Comment by Swatinem** [comment]
I would go with messages/sec, as that is pretty much the metric we look for in prod.
However those few messages we have there to test probably do not fully reflect whatever we see in prod though.

**Review by Swatinem — APPROVED** [review]

**Review by untitaker — APPROVED** [review]

**Pull request merged**